src/sdam/topology.ts

```diff
@@ -457,8 +459,17 @@ export class Topology extends TypedEventEmitter<TopologyEvents> {
       }
     }
 
+    const timeoutMS = this.client.options.timeoutMS;
+    const timeout = timeoutMS != null ? Timeout.expires(timeoutMS) : undefined;
+    const serverSelectionTimeoutMS = this.s.serverSelectionTimeoutMS;
+    const serverSelectionTimeout =
+      timeoutMS != null ? Timeout.expires(csotMin(timeoutMS, serverSelectionTimeoutMS)) : undefined;
```

Comment:
We can also just  do 
```
serverSelectionTimeout  = Timeout.expires(csotMin(timeoutMS, serverSelectionTimeoutMS)
```
and change `csotMin` to allow and detect undefined parameters and return the other parameter, if both are undefined, return 0 (CSOT's Infinity)

```
export function csotMin(duration1: number, duration2: number): number {
  if (duration1 == null && duration2 == null) return 0;
  if (duration1 == null || duration1 === 0) return duration2;
  if (duration2 == null || duration2 === 0) return duration1;
    return Math.min(duration1, duration2);
}
```

Just a suggestion, feel free to ignore, I just feel it would mean less checking for undefined variables in the future.